### ISSUE TITLE
Errors silenced when one asyncgenerator calls another

### ISSUE THREAD

**Issue by dfabulich** [bikeshaving/crank#77]
```html
<html>
<body>
<script type=module>
import {createElement} from "https://unpkg.com/@bikeshaving/crank@0.1.1?module";
import {renderer} from "https://unpkg.com/@bikeshaving/crank@0.1.1/dom?module";

async function* Two() {
  console.log(undeclared);
  return "hi";
}

async function* One() {
  for await (let props of this) yield createElement(Two, props);
}

void async function() {
  await renderer.render(createElement(One), document.body);
}();
</script>
</body>
</html>
```

Expected: This code should log an error on the console, "ReferenceError: undeclared is not defined"

Actual: Nothing appears on the console. Pledge calls `.catch` on Two's Promise and ignores the error.

(This wasted a _lot_ of my time this evening. 😖)

**Comment by dfabulich** [comment]
This is no better. None of the catch blocks are reached; the error just vanishes into thin air.

```html
<html>
<body>
<script type=module>
import {createElement} from "https://unpkg.com/@bikeshaving/crank@0.1.1?module";
import {renderer} from "https://unpkg.com/@bikeshaving/crank@0.1.1/dom?module";

async function* Two() {
  console.log(undeclared);
  return "hi";
}

async function* One() {
  try {
    for await (let props of this) yield createElement(Two, props);
  } catch (e) {
    console.error(e);
  }
  
}

void async function() {
  try {
    await renderer.render(createElement(One), document.body);
  } catch (e) {
    console.error(e);
  }
}();
</script>
</body>
</html>
```

**Comment by brainkim** [comment]
@dfabulich Sorry to hear that! Investigating.

**Comment by brainkim** [comment]
Just updating. This is currently priority one after spending most of the day solving easier bugs. Feel free to mute if you’re not interested.

I think I almost understand the bug. My current theory is that the error is attempting to propagate up the tree, but because each call to async generators happens in sequence, there is a possibility of a deadlock where the async generator component is waiting for something to happen (in this case, the new props) and unable to respond to the thrown error. The solution would be to make sure the async iterator unblocks so the error can be thrown in, but I wonder what would happen if the async generator is blocked on a different promise. Difficult to solve and wondering if it’s one of those intractable problems with the design of async generator functions. Deadlocks are super easy and I wish they designed it a little better. Will continue playing around to prove my theory and think about solutions for mitigation but right now my brain is fried.

**Comment by brainkim** [comment]
My theory was correct and this should be fixed in 0.1.2. There are still lingering concerns about async generator components deadlocking and preventing errors from being propagated, but this fix should handle the simple case of when the component is awaiting new props. Overall the error handling code could be more rigorously tested and specified, and I’ll make sure this happens soon.
